examples/multiple_systems_update.py

```diff
@@ -0,0 +1,58 @@
+# Example of updating with different H,y etc
+
+from simdkalman import KalmanFilter
+import numpy as np 
+
+# Process noise 
+Q = np.zeros((2,3,3))
+Q[0,:,:] = np.eye(3)
+Q[1,:,:] = np.eye(3)
+
+
+# Prior state means 
+m0 = np.zeros((2,3,1))
+m0[0,:,:] = np.array([[1.5, 1.5, 1.5]]).transpose()
+m0[1,:,:] = np.array([[1.5, 1.5, 1.5]]).transpose()
+
+# Prior covariances
+P0 = np.zeros((2, 3, 3))
+P0[0, :, :] = np.eye(3)
+P0[1, :, :] = 0.5 * np.eye(3)
+
+# Observations
+y_update = np.zeros((2, 1, 1))
+y_update[0, :, :] = np.array([[-1.4]])
+y_update[1, :, :] = np.array([[-0.33]])
+
+# Measurement var
+R = np.zeros((2, 1, 1))
+R[0, :, :] = np.eye(1)
+R[1, :, :] = 0.66*np.eye(1)
+
+# Observation equations 
+H0 = np.array([[0.8, 0.2, 0]])
+H1 = np.array([[0.9, 0.1, 0]])
+H0.shape = (1, 3)
+H1.shape = (1, 3)
+H = np.zeros((2, 1, 3))
+H[0, :, :] = H0
+H[1, :, :] = H1
+
+# Transition(s) 
+testing_different_transitions = False
+A0 = np.array([[0.5, 0.3, 0.2], [1, 0, 0], [0, 1, 0]])
+if testing_different_transitions:
+    A = np.zeros((2,3,3) )
+    A1 = np.array([[0.4, 0.4, 0.2], [1, 0, 0], [0, 1, 0]])
+    A[0,:,:] = A0
+    A[1,:,:] = A1
+else:
+    A = A0
+
+kf = KalmanFilter(
+    state_transition=A,  # A
+    process_noise=Q,  # Q
+    observation_model=H,  # H
+    observation_noise=R)  # R
+m1, P1, K, ll = kf.update(m0,P0,y_update, log_likelihood=True)
```

Comment:
There might be a small misunderstanding here and I think this example should be either edited a bit or removed: In this version, the state transition _A_ is not used at all. The reason is that a call to `kf.predict_next` is missing.

This is admittedly really confusing, but the reasoning behind the terminology is this: there are two basic KF operations: the prediction step (which uses the matrices _A_ & _Q_) and the update step (using matrices _H_ & _R_). `kf.update` is the latter step and the former is called `kf.predict_next`. The confusing part, in retrospect, is that there is also `kf.predict` which is a "higher-level" method.

I would do one of the following:
 * remove this example entirely
 * add `predict_next` and rename to something like `single_step_multiple_systems.py`. It could also make sense to migrate to use the "primitives" module, since if the matrices do not change over time, it is perhaps less common to have to use the single-step versions